roedoejet/g2p
# Dev.ej/lexicon tokenizer

### PR Goal? 

Introduce the lexicon tokenizer

### Fixes? 

Fixes #401

### Feedback sought? 

This is a draft PR, seeking early feedback on the algorithm I implemented.

### Priority? 

medium-high

### Tests added? 

yes

### How to test? 

Run `g2p convert "We'll let go, she'll stop." eng eng-ipa` and see the output `wil lɛt ɡoʊ, ʃil stɑp.` instead of `wi' lɛt ɡoʊ, ʃi' stɑp.`

### Confidence? 

moderate only.

 - unit testing tells me it's fine
 - but I have not yet done any benchmarking to see if we need to optimize

### Version change? 

Probably yes, at least a patch bump, since this is a bug fix, maybe even a minor bump?
